src/test/java/org/generationcp/breeding/manager/listmanager/GermplasmColumnValuesGeneratorTest.java

```diff
@@ -605,4 +607,91 @@ public void testSetImmediateSourcePreferredNameColumnValues() {
+			Mockito.verify(this.fillColumnSource).setColumnValueForItem(itemId, columnName, "2");
+		}
+	}
+
+	@Test
+	public void testSetCrossMaleInfoColumnValuesWithParentNAME() {
+		List<Germplasm> germplasms = this.generateListofGermplasm(true, 1, 2);
+		ArrayList<Integer> parent = new ArrayList<>();
+		parent.add(2);
+
+		final Map<Integer, String> namesMap = this.generateGIDStringMap("ABCDEFG",parent);
+		Mockito.when(this.germplasmDataManager.getPreferredNamesByGids(ArgumentMatchers.anyListOf(Integer.class))).thenReturn(namesMap);
+		Mockito.doReturn(germplasms).when(this.germplasmDataManager)
+				.getGermplasms(GermplasmColumnValuesGeneratorTest.GID_LIST);
+		final String columnName = ColumnLabels.PARENTAGE.getName();
+		this.valuesGenerator.setCrossMalePrefNameColumnValues(columnName);
+//		for (final Object itemId : GermplasmColumnValuesGeneratorTest.ITEMS_LIST) {
+//			Mockito.verify(this.fillColumnSource).setColumnValueForItem(itemId, columnName, "-");
+//		}
+
+		for(final Object itemId : parent)  {
+			this.fillColumnSource.setColumnValueForItem(itemId, columnName, "ABCDEFG 1");
+		}
+	}
+
+	@Test
+	public void testSetCrossFemaleInfoColumnValuesWithParentNAME() {
+		List<Germplasm> germplasms = this.generateListofGermplasm(true, 1, 2);
+		ArrayList<Integer> parent = new ArrayList<>();
+		parent.add(1);
+
+		final Map<Integer, String> namesMap = this.generateGIDStringMap("ABCDEFG",parent);
+		Mockito.when(this.germplasmDataManager.getPreferredNamesByGids(ArgumentMatchers.anyListOf(Integer.class))).thenReturn(namesMap);
+		Mockito.doReturn(germplasms).when(this.germplasmDataManager)
+				.getGermplasms(GermplasmColumnValuesGeneratorTest.GID_LIST);
+		final String columnName = ColumnLabels.PARENTAGE.getName();
+		this.valuesGenerator.setCrossFemaleInfoColumnValues(columnName, FillWithOption.FILL_WITH_CROSS_FEMALE_NAME);
+		for (final Object itemId : GermplasmColumnValuesGeneratorTest.ITEMS_LIST) {
+			Mockito.verify(this.fillColumnSource).setColumnValueForItem(itemId, columnName, "-");
+		}
+
+		for(final Object itemId : parent)  {
+			this.fillColumnSource.setColumnValueForItem(itemId, columnName, "ABCDEFG 2");
+		}
+	}
+
+	private List<Germplasm> generateListofGermplasm(final boolean isDerivative, Integer femaleParent, Integer maleParent) {
+		final List<Germplasm> list = new ArrayList<>();
+		for (final Integer gid : GermplasmColumnValuesGeneratorTest.GID_LIST) {
+			final Germplasm germplasm = new Germplasm();
+			germplasm.setGid(gid);
+			if (isDerivative) {
+				germplasm.setGnpgs(-1);
+				germplasm.setGpid1(femaleParent);
+				germplasm.setGpid2(maleParent);
+				germplasm.setFemaleParentPreferredName(String.valueOf(femaleParent));
+				germplasm.setMaleParentPreferredName(String.valueOf(maleParent));
+			} else {
+				germplasm.setGnpgs(2);
+				germplasm.setGpid1(femaleParent);
+				germplasm.setGpid2(maleParent);
+				germplasm.setFemaleParentPreferredName(String.valueOf(femaleParent));
```

Comment:
Also I would suggest to use a different value for parent preferred name instead of the just the GID. you can append a prefix to the parent GID to use as preferred name. So that assertions will be stronger